ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py

```diff
@@ -875,8 +851,9 @@ def fetch_snapshots_for_all_subset(subset, client_obj):
                 if subset['limit'] is not None and total_snap.__len__() >= subset['limit']:
                     total_snap = total_snap[0:subset['limit']]
                     break
-        result['snapshots'] = generate_dict('snapshots', total_snap)['snapshots']
-        return result
+        if total_snap.__len__() > 0:
```

Comment:
this is fix for 145.. there can be a case that a volume has no snapshots at all.. hence no point in calling generate_dict in that case